domains/v1/domains_test.go

```diff
@@ -4,52 +4,90 @@ import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gmeasure"
 )
 
 var _ = Describe("domains", func() {
 	Describe("GET /v3/domains", func() {
-		Measure("as admin", func(b Benchmarker) {
-			workflowhelpers.AsUser(testSetup.AdminUserContext(), testConfig.BasicTimeout, func() {
-				helpers.TimeCFCurl(b, testConfig.BasicTimeout, "/v3/domains")
-			})
-		}, testConfig.Samples)
 
-		Measure("as regular user", func(b Benchmarker) {
-			workflowhelpers.AsUser(testSetup.RegularUserContext(), testConfig.BasicTimeout, func() {
-				helpers.TimeCFCurl(b, testConfig.BasicTimeout, "/v3/domains")
-			})
-		}, testConfig.Samples)
+		It("gets /v3/domains as admin efficiently", func() {
+			experiment := gmeasure.NewExperiment("as admin")
+			AddReportEntry(experiment.Name, experiment)
 
-		Measure(fmt.Sprintf("as admin with page size %d", testConfig.LargePageSize), func(b Benchmarker) {
-			workflowhelpers.AsUser(testSetup.AdminUserContext(), testConfig.LongTimeout, func() {
-				helpers.TimeCFCurl(b, testConfig.LongTimeout, fmt.Sprintf("/v3/domains?per_page=%d", testConfig.LargePageSize))
-			})
-		}, testConfig.Samples)
+			experiment.Sample(func(idx int) {
+				experiment.MeasureDuration("GET /v3/domains", func() {
+					workflowhelpers.AsUser(testSetup.AdminUserContext(), testConfig.BasicTimeout, func() {
+						helpers.TimeCFCurl(testConfig.BasicTimeout, "/v3/domains")
+					})
+				})
+			}, gmeasure.SamplingConfig{N: testConfig.Samples})
+		})
+
+		It("gets /v3/domains as a regular user efficiently", func() {
+			experiment := gmeasure.NewExperiment("as user")
+			AddReportEntry(experiment.Name, experiment)
+
+			experiment.Sample(func(idx int) {
+				experiment.MeasureDuration("GET /v3/domains", func() {
+					workflowhelpers.AsUser(testSetup.RegularUserContext(), testConfig.BasicTimeout, func() {
+						helpers.TimeCFCurl(testConfig.BasicTimeout, "/v3/domains")
+					})
+				})
+			}, gmeasure.SamplingConfig{N: testConfig.Samples})
+		})
+
+		It(fmt.Sprintf("gets /v3/domains as admin with page size %d efficiently", testConfig.LargePageSize), func() {
+			experiment := gmeasure.NewExperiment(fmt.Sprintf("as admin with page size %d", testConfig.LargePageSize))
+			AddReportEntry(experiment.Name, experiment)
+
+			experiment.Sample(func(idx int) {
+				experiment.MeasureDuration(fmt.Sprintf("GET /v3/domains"), func() {
+					workflowhelpers.AsUser(testSetup.AdminUserContext(), testConfig.LongTimeout, func() {
+						helpers.TimeCFCurl(testConfig.LongTimeout, fmt.Sprintf("/v3/domains?per_page=%d", testConfig.LargePageSize))
+					})
+				})
+			}, gmeasure.SamplingConfig{N: testConfig.Samples})
+		})
 	})
 
 	Describe("GET /v3/organizations/:guid/domains", func() {
-		Measure("as admin", func(b Benchmarker) {
+		It("gets /v3/organizations/:guid/domains as admin efficiently", func() {
 			orgGUIDs := helpers.GetGUIDs(testSetup.AdminUserContext(), testConfig, "/v3/organizations")
```

Comment:
There's a lot of repetition in this file with just a few strings varying between the different tests. Could you just loop over a map of those and chop this to a fraction of its current size?

Reply:
A helper function might be better/easier than a bunch of loops actually.